ValidationLibrary.Tests/RepositoryValidatorTests.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using NUnit.Framework;
+using Octokit;
+
+namespace ValidationLibrary.Tests
+{
+    [TestFixture]
+    public class RepositoryValidatorTests
+    {
+        private const string ConfigFileName = "repository-validator.json";
+
+        private RepositoryValidator _repositoryValidator;
+
+        private IValidationRule[] _mockRules;
+        private IGitHubClient _mockClient;
+        //private IRepositoriesClient _mockRepositoriesClient;
+        private IRepositoryContentsClient _mockRepositoryContentsClient;
+
+        [SetUp]
+        public void SetUp()
+        {
+
+            var mockLogger = Substitute.For<ILogger<RepositoryValidator>>();
+
+            _mockClient = Substitute.For<IGitHubClient>();
+
+            _mockRules = new IValidationRule[]
+            {
+                Substitute.For<IValidationRule>(),
+                Substitute.For<IValidationRule>(),
+                Substitute.For<IValidationRule>()
+            };
+
+            _repositoryValidator = new RepositoryValidator(mockLogger, _mockClient, _mockRules);
+
+            //_mockRepositoriesClient = Substitute.For<IRepositoriesClient>();
+            _mockRepositoryContentsClient = Substitute.For<IRepositoryContentsClient>();
+            _mockClient.Repository.Content.Returns(_mockRepositoryContentsClient);
+        }
+
+        [Test]
+        public async Task Init_InitiatesAllRules()
+        {
+            await _repositoryValidator.Init();
+
+            foreach (var mockRule in _mockRules)
+            {
+                await mockRule.Received().Init(_mockClient);
+            }
+        }
+
+        [Test]
+        public async Task Validate_ValidatesWithAllRulesIfNoneAreIgnoredAndConfigIsNotFound()
+        {
+            _mockRepositoryContentsClient
+                .GetAllContents("testOwner", "mock-repositry", ConfigFileName)
+                .Returns<Task<IReadOnlyList<RepositoryContent>>>(x => { throw new NotFoundException("no message", HttpStatusCode.NotFound); });
+
+            var repository = CreateRepository("testOwner", "mock-repositry", true, false);
+            var result = await _repositoryValidator.Validate(repository, false);
+            Assert.NotNull(result);
```

Comment:
I would recommend FluentAssertions.

Reply:
Could also just add some air if you are not using the // Arrange, // Act // Assert comments. Just to make it easier to locate the method call under test.

Reply:
Also just any non-null result is OK? I doubt it.

Reply:
The call results in an instance where Results property has a length of 3, but all items are null. Seems fishy.

Reply:
I agree, this was a bit lazy.